apps/bug-cd/src/app/main-navigation.component.ts

```diff
@@ -40,25 +41,19 @@ export class NavigationComponent {
 @Component({
   standalone: true,
   imports: [NavigationComponent, NgIf, AsyncPipe],
-  template: `
-    <ng-container *ngIf="info$ | async as info">
-      <ng-container *ngIf="info !== null; else noInfo">
-        <app-nav [menus]="getMenu(info)" />
-      </ng-container>
-    </ng-container>
-
-    <ng-template #noInfo>
-      <app-nav [menus]="getMenu('')" />
-    </ng-template>
-  `,
+  template: ` <app-nav [menus]="(menus$ | async)!" /> `,
   host: {},
 })
 export class MainNavigationComponent {
   private fakeBackend = inject(FakeServiceService);
+  menus$!: Observable<MenuItem[]>;
+  constructor() {
+    this.menus$ = this.fakeBackend
+      .getInfoFromBackend()
+      .pipe(map((info) => this.getMenu(info || '')));
+  }
```

Comment:
This is better but why do you reassign your obs inside the constructor?

Assign it directly when you declare it.

Something you need to reassign it inside the ngoninit method because you need some inputs, but when you don't need any, no need to do it like it. 

it would be easier to read and you will not need to declare a type. Type will be infered by typescript directly

Reply:
Absolutely, forgot to move it when refactoring, now pushed